src/core/configs/cluster.rs

```diff
@@ -0,0 +1,397 @@
+            return result;
+        }
+        result = self.aggregate(&self.vdisks);
+        if result.is_some() {
+            debug!("some vdisks in config are invalid");
+            return result;
+        }
+
+        if self
+            .vdisks
+            .as_ref()?
+            .iter()
+            .group_by(|x| x.id)
+            .into_iter()
+            .map(|(_, group)| group.count())
+            .filter(|x| *x > 1)
+            .count()
+            != 0
+        {
+            debug!("config contains duplicates vdisks ids");
+            return Some("config contains duplicates vdisks ids".to_string());
+        }
+        if self
+            .nodes
+            .as_ref()?
+            .iter()
+            .group_by(|x| x.name.clone())
+            .into_iter()
+            .map(|(_, group)| group.count())
+            .filter(|x| *x > 1)
+            .count()
+            != 0
+        {
+            debug!("config contains duplicates nodes names");
+            return Some("config contains duplicates nodes names".to_string());
+        }
+
+        for node in self.nodes.as_ref()?.iter() {
+            node.prepare();
+        }
+
+        for vdisk in self.vdisks.as_ref()?.iter() {
+            for replica in vdisk.replicas.as_ref()?.iter() {
+                match self.nodes.as_ref()?.iter().find(|x| x.name == replica.node) {
+                    Some(node) => {
+                        if node.disks.as_ref()?.iter().find(|x| x.name == replica.disk) == None {
+                            debug!(
+                                "cannot find in node: {}, disk with name: {} for vdisk: {}",
+                                replica.node.as_ref()?,
+                                replica.disk.as_ref()?,
+                                vdisk.id?
+                            );
+                            return Some(format!(
+                                "cannot find in node: {}, disk with name: {} for vdisk: {}",
+                                replica.node.as_ref()?,
+                                replica.disk.as_ref()?,
+                                vdisk.id?
+                            ));
+                        }
+                    }
+                    None => {
+                        debug!(
+                            "cannot find node: {} for vdisk: {}",
+                            replica.node.as_ref()?,
+                            vdisk.id?
+                        );
+                        return Some(format!(
+                            "cannot find node: {} for vdisk: {}",
+                            replica.node.as_ref()?,
+                            vdisk.id?
+                        ));
+                    }
+                }
+            }
+        }
+
+        None
+    }
+}
+
+use crate::core::data::Node as DataNode;
+use crate::core::data::NodeDisk as DataNodeDisk;
+use crate::core::data::VDisk as DataVDisk;
+
+use std::collections::HashMap;
+
+pub trait BobConfig {
```

Comment:
Why trait and empty struct?
Isn't it better to implement same methods for Cluster?
With usage:
```rust 
let cluster_config = Cluster::from_file(filename).unwrap();
let vdisk_data: Vec<_> = cluster_config.to_data().unwrap();
```